plugin/evm/vm_test.go

```diff
@@ -2110,3 +2088,149 @@ func TestCreateHandlers(t *testing.T) {
 		require.ErrorIs(t, elem.Error, rpc.ErrMissingBatchResponse)
 	}
 }
+
+// TestGraniteDeactivatesBlockGasCost verifies block gas cost semantics across
+// the Fortuna -> Granite fork boundary:
+//   - Under Fortuna, attempting to build a second block too quickly should fail
+//     with ErrInsufficientBlockGas.
+//   - Under Granite, block gas cost is disabled, building succeeds and the
+//     header extra's BlockGasCost is 0.
+//
+// This guards against regressions in block gas charging behavior after Granite.
+func TestGraniteDeactivatesBlockGasCost(t *testing.T) {
```

Comment:
(perhaps an unpopular opinion) I'm not really convinced that we need these integration tests for this.

These tests are really hard to setup and pretty annoying to maintain... What coverage are we realistically getting that we aren't getting from the `block_gas_cost_test.go` file?